Add ssl options to webclient in oidc-client

sd-f opened the pull request:
Fixes #14480

Commit:
Add ssl options to webclient in oidc-client

sberyozkin commented on extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java:
```diff
@@ -147,6 +149,31 @@ public OidcClient apply(String tokenRequestUri) {
         });
     }
 
+    private static void setHttpsClientOptions(URI authServerUri, WebClientOptions options) {
```
@sd-f can you please rename it to `setWebClientOptions` ? It is not only https properties that are set there, right now `trustAll` is dealt in `OidcCommonUtils`...

sd-f replied:
@sberyozkin done :)

sberyozkin commented:
@sd-f I think we also need to update the oidc client test to use HTTPS; would you like to try it ? 
Apply the same changes as [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24) (the same property is referenced twice), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24), [here](https://github.com/quarkusio/quarkus/blob/master/integration-tests/oidc/src/test/java/io/quarkus/it/keycloak/KeycloakRealmResourceManager.java#L24)(as well as the https port a few lines below) and finally [here](https://github.com/quarkusio/quarkus/blob/0f33c23aac875295693d00cef073adabbff8d780/integration-tests/oidc/src/main/resources/application.properties#L2) (see also a trust all property) to `integration-tests/oidc-client`.

Give it a try please if you don't mind, I can help as well

sd-f commented:
> 
> Give it a try please if you don't mind, I can help as well

I'll try

sberyozkin commented:
@sd-f thanks

sd-f commented:
> @sd-f thanks

I am not sure why but it worked without `quarkus.oidc.tls.verification=none` or `trust-all`
I added
```
    static {
        RestAssured.useRelaxedHTTPSValidation();
    }
```
and this seems to be enouth. 

I also added the https endpoint to the keycloak docker container in maven.

And finally I hope my formatter worked xD

sberyozkin commented:
@sd-f Great,  Re the import order, that should be the same after building the source (I've tested it in some other file), did you rebuild `oidc-client/runtime` :-) ?
Please also squash the commits

sd-f commented:
> @sd-f Great, Re the import order, that should be the same after building the source (I've tested it in some other file), did you rebuild `oidc-client/runtime` :-) ?
> Please also squash the commits

Thank your for your help, i followed the contribution guide and imported "import order config" for intellij after that the order was way different. mvn checks and verify is not complaining (formatter, ismimport plugin)

sberyozkin commented:
@sd-f Np, thanks; `Initial JDK 11 Build` is failing right now with:
```
Error:  Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java
```

What happens if you just do a clean rebuild on the cmd line ? I think it should restore them; if it does not - simply restore the original order for now - may be the import order config is not effective for IntelliJ ... CC @gsmet

sd-f commented:
> @sd-f Np, thanks; `Initial JDK 11 Build` is failing right now with:
> 
> ```
> Error:  Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java -> [Help 1]
> org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java
> ```
> 
> What happens if you just do a clean rebuild on the cmd line ? I think it should restore them; if it does not - simply restore the original order for now - may be the import order config is not effective for IntelliJ ... CC @gsmet

Thanks, already fixed, I did the imports by hand. impsort:check is successful and I did a fresh build on command line before commit :)

sd-f commented:
> @sd-f Np, thanks; `Initial JDK 11 Build` is failing right now with:
> 
> ```
> Error:  Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java -> [Help 1]
> org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.4.1:check (check-imports) on project quarkus-oidc-client: Imports are not sorted in /home/runner/work/quarkus/quarkus/extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java
> ```
> 
> What happens if you just do a clean rebuild on the cmd line ? I think it should restore them; if it does not - simply restore the original order for now - may be the import order config is not effective for IntelliJ ... CC @gsmet

sberyozkin commented:
@sd-f cool; thanks sorting it out :-)

sberyozkin commented:
@sd-f OK, the builds are good, so lets merge it. 
Can you please give me a favour and check sometime later how would using `.getAbs()` (vs `.get()`) change the picture. @vsevel uses `.getAbs()` in his PR.
The question I have, what is cheaper. If set here the default host and port then may be it is more efficient to use only the path component of the auth server url and also token url ? At the moment we set the default host/port - but also use absolute URIs with the `.get()` and `post()`. It feels like we can further optimize something :-). I'll open an issue and CC you

Thanks for your contribution

rsvoboda commented:
@gsmet fyi, this should get into 1.11.1.
